### ISSUE TITLE
Using boost::filesystem in Boost 1.75.0 onwards fails trying to call syscall(__NR_statx, ...)

### ISSUE THREAD

**Issue by fun4jimmy** [microsoft/WSL#6994]
### Windows Build Number

Microsoft Windows [Version 10.0.19042.985]

### WSL Version

- [ ] WSL 2
- [X] WSL 1

### Kernel Version

Linux version 4.4.0-19041-Microsoft (Microsoft@Microsoft.com) (gcc version 5.4.0 (GCC) ) #488-Microsoft Mon Sep 01 13:43:00 PST 2020

### Distro Version

Ubuntu 18.04

### Other Software

gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Boost 1.75.0

### Repro Steps

Download, extract and build Boost 1.75.0:
```bash
wget https://boostorg.jfrog.io/artifactory/main/release/1.75.0/source/boost_1_75_0.tar.gz
tar -zxf boost_1_75_0.tar.gz
cd boost_1_75_0
./bootstrap.sh --prefix=installed
# This call to b2 will perform some configuration checks including the statx syscall check which will pass successfully
./b2 install variant=debug --without-python --without-iostreams --without-mpi
```

Write a simple program `status_test.cpp` that uses a `boost::filesystem` call that relies on statx:
```cpp
#include <boost/filesystem/operations.hpp>

int main(int argc, char** argv)
{
  boost::filesystem::path FilePath("/etc/fstab");
  auto Status = boost::filesystem::status(FilePath);
  return 0;
}
```

Compile and run the program (assumes status_test.cpp is in the same directory as the build of boost):
```bash
g++ status_test.cpp -g -Iinstalled/include -Linstalled/lib -lboost_filesystem
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/installed/lib
./a.out
```

### Expected Behavior

The call to `boost::filesystem::status` in the sample application was successful and the program returns 0 with no errors.

### Actual Behavior

The actual output of the program is a thrown exception with a misleading error message:
```bash
terminate called after throwing an instance of 'boost::filesystem::filesystem_error'
  what():  boost::filesystem::status: Function not implemented: "/etc/fstab"
Aborted (core dumped)
```

If you step into the call to `boost::filesystem::status` you will see that the program fails when trying to call the boost::filesystem statx wrapper (boost_1_75_0\libs\filesystem\src\operations.cpp(367)):
```cpp
inline int statx(int dirfd, const char* path, int flags, unsigned int mask, struct ::statx* stx) BOOST_NOEXCEPT
{
  return ::syscall(__NR_statx, dirfd, path, flags, mask, stx);
}
```

I think the problem is that the WSL kernel headers indicate that `syscall(__NR_statx, ...)` will work but the function is not actually implemented.

**Comment by fun4jimmy** [comment]
I think this may be a similar issue found by nodejs https://github.com/nodejs/node/issues/30860.
This WSL issue may also be related https://github.com/microsoft/WSL/issues/3023.

**Comment by fun4jimmy** [comment]
It looks like the authors of boost::filesystem are adding a workaround to manually disable statx support if necessary (https://github.com/boostorg/filesystem/blob/develop/build/Jamfile.v2#L66) but that is not available in any released version of boost (1.75.0 or 1.76.0) yet.
Also, it would be preferable to be able to use the same boost binary on WSL and native Ubuntu distributions with performance benefits from using statx.
